### PR TITLE
Active Users Component

### PR DESCRIPTION
**Pull request by suvasan19**

### CODE REVIEW

**Comment by krsyoung** [comment]
@suvasan19 I think we have an initial version here. Definitely need some polishing ... take a look at the code changes and let me know what you think.

We should probably include the following in the README as well:

1. Dependency (soft) on Bootstrap 5.x
2. Current requirement to load the gem directly from our GitHub (branch can vary)
3. Need to investigate slots (@jonathanloos can probably help) for custom HTML
4. Need to compare this to the webcomponents version per @Technickel-Dev

File: app/components/matey/active_users_component.html.erb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      <div id="top half col" class="row">
+        <div class="col m-2">
+          <h1 class="card-title"><%= @current_period.to_s %></h5>
+        </div>
+        <div class="col">
+          <p class="card-text"><%= (@change_active_number > 0 ? "+" : "") + @change_active_number.to_s + " / " + @change_active_percent.to_s + "%" %></p>
+        </div>
+      </div>
+      <div class="col">
+        <h5 class="card-text my-2"><strong> Active Users </strong> since <%= time_ago_in_words(@time_window.ago) %> ago</h6>
```
Comment (Technickel-Dev) [inline app/components/matey/active_users_component.html.erb:13]: Start as h5, but close h6

File: app/components/matey/active_users_component.html.erb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<div class="card widget-flat shadow-sm">
+  <div class="card-body">
+    <div id="first col" class="col">
+      <div id="top half col" class="row">
+        <div class="col m-2">
+          <h1 class="card-title"><%= @current_period.to_s %></h5>
```
Comment (Technickel-Dev) [inline app/components/matey/active_users_component.html.erb:6]: Start as h1 but closes h5

**Review by Technickel-Dev — CHANGES_REQUESTED** [review]
Just two small fixes in the HTML to avoid bugs in the future

File: app/components/matey/active_users_component.html.erb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<div class="card widget-flat shadow-sm">
```
Comment (krsyoung) [inline app/components/matey/active_users_component.html.erb:1]: @suvasan19 what's going on with the classes here? If we're introducing some weird classes?

What is "widget-flat"?

File: app/components/matey/active_users_component.html.erb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  <div class="card-body">
+    <div id="first col" class="col">
```
Comment (krsyoung) [inline app/components/matey/active_users_component.html.erb:3]: I'm not sure if I added this, but what is the purpose of the ids? I'm sure we don't need them .. (it won't work if we have multiple cards on the same page as ids must be unique).

File: app/components/matey/active_users_component.html.erb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+      <div id="top half col" class="row">
+        <div class="col m-2">
+          <h1 class="card-title"><%= @current_period.to_s %></h1>
+        </div>
+        <div class="col">
+          <p class="card-text"><%= (@change_active_number > 0 ? "+" : "") + @change_active_number.to_s + " / " + @change_active_percent.to_s + "%" %></p>
+        </div>
+      </div>
+      <div class="col">
```
Comment (krsyoung) [inline app/components/matey/active_users_component.html.erb:12]: Is this `col` in the `row`? Seems like it is outside?

**Comment by krsyoung** [comment]
@suvasan19 I've found a number of issues with the HTML, which I'm not sure if there is a reason for (or if I was the one that added them!). Let me know if they are valid for some reason, or let's clean them up.

**Pull request merged**